src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -2076,9 +2076,9 @@ void primitive_inst::add_profiling_data(instrumentation::pipeline_stage stage, b
 #ifdef GPU_DEBUG_CONFIG
     int64_t curr_iter = -1;
     GPU_DEBUG_IF(debug_config->dump_prof_data_iter_params.is_enabled) {
-        curr_iter = get_network().get_current_iteration_num();
+        curr_iter = get_network().get_current_iteration_num() - 1;
```

Comment:
I think we'd better to fix get_current_iteration_num() itself to return the expected iteration number, instead of fixing after calling it.

Reply:
Fixed behavior for increasing network's iteration itself